app/assets/javascripts/discourse/app/controllers/full-page-search.js

```diff
@@ -464,9 +499,14 @@ export default Controller.extend({
     });
   },
 
+  @action
+  recheckSearchResults() {
+    this.notifyPropertyChange("searchResultPosts");
+  },
+
```

Comment:
Instead of having to do:

```js
     api.addAdditionalSearchResults(AIResults);
     this.args.outletArgs.recheckSearchResults();
```

inside Discourse AI plugin to refresh search results, is there an easy way to automatically trigger `recheckSearchResults` in core whenever the API method is called?

I suppose we can use `getOwner`, `getController` and call the method after the pluginAPI method is called, but it feels like it introduces code smell. Though, I still feel this recheckingSearchResults should be automatically handled by the API rather than having to be invoked in the plugin.

Reply:
Would it work to do something like: 

```
  @discourseComputed("model.posts", "additionalSearchResults")
  searchResultPosts(posts) {
```

So basically add a way to watch the list of additional search results and recalculate `searchResultPosts` whenever that changes.

Reply:
This doesn't seem to work. I think it's not working because `additionalSearchResults` is defined outside of the controller so that it can be used along with the exported plugin API function:

https://github.com/discourse/discourse/blob/639b88f07ea96fd76c86592e1d1d7ad001d51e26/app/assets/javascripts/discourse/app/controllers/full-page-search.js#L47

https://github.com/discourse/discourse/blob/639b88f07ea96fd76c86592e1d1d7ad001d51e26/app/assets/javascripts/discourse/app/controllers/full-page-search.js#L57-L59

Reply:
I'm going to go ahead and just solve this by doing:

```diff
  /**
   * Allows you add additional search results to the full page search results.
   * Search results added through this API will be mixed with initial search results
   * and ranked using the reciprocal ranking fusion algorithm.
   *
   * (See `reciprocallyRankedList()` in `discourse/app/lib/search.js` for more
   * details on how reciprocal ranking works)
   *
   *
   * Example:
   *
   * api.addSearchResults(newPostArray);
   */
  addSearchResults(results) {
    additionalResults(results);
+    this.container.lookup("controller:full-page-search").recheckSearchResults();
  }
```

Not sure if this is the best solution, but it works for now. We can always revisit and update the implementation in a separate PR if we can find a better solution.